react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecText.java

```diff
@@ -571,4 +577,40 @@ private boolean shouldDelegateTextChangeCalls() {
     private boolean isPreTag() {
         return PRE_TAG.equals(mTagName);
     }
+
+    public void setCursorColor(@Nullable Integer color) {
+        // This is nearly a direct copy of the ReactNative implementation:
+        // https://github.com/facebook/react-native/blob/233fdfc014bb4b919c7624c90e5dac614479076f/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java#L422-L457
+        // API 29 supports setTextCursorDrawable which would be a cleaner way to handle this when
+        // an upgrade to that API level occurs.
+        try {
+            // Get the original cursor drawable resource.
+            Field cursorDrawableResField = TextView.class.getDeclaredField("mCursorDrawableRes");
+            cursorDrawableResField.setAccessible(true);
+            int drawableResId = cursorDrawableResField.getInt(this);
+
+            // The view has no cursor drawable.
+            if (drawableResId == 0) {
+                return;
+            }
+
+            Drawable drawable = ContextCompat.getDrawable(getContext(), drawableResId);
+            if (color != null) {
+                drawable.setColorFilter(color, PorterDuff.Mode.SRC_IN);
+            }
+            Drawable[] drawables = {drawable, drawable};
+
+            // Update the current cursor drawable with the new one.
+            Field editorField = TextView.class.getDeclaredField("mEditor");
+            editorField.setAccessible(true);
+            Object editor = editorField.get(this);
+            Field cursorDrawableField = editor.getClass().getDeclaredField("mCursorDrawable");
+            cursorDrawableField.setAccessible(true);
+            cursorDrawableField.set(editor, drawables);
```

Comment:
I am not understanding the logic here. Why do we need to set accessible=true in multiple places? Is that necessary to be able to set cursor color? @chipsnyder

Reply:
This method is a bit of a hack to get to the desired goal, but I can add some more info. In Android, the cursor color is managed by the theme and not on a control by control basis. You can set it in XML via [android:textCursorDrawable](https://developer.android.com/reference/android/widget/TextView.html#attr_android:textCursorDrawable). However there isn't an officially supported way to do change this programmatically in API 28 and below. 

In API 27 and below, you can use reflection to dynamically lookup the drawable for the current TextView and replace it with a new drawable. Setting `setAccessible` to true allows us to make the changes we need through reflection. Without that set to true, the app will throw `IllegalAccessException` 

Here's a breakdown of where we set it. 

| Variable | How we use `setAccessible` | 
| ---- | ---- |
| cursorDrawableResField | Allows the lookup of`drawableResId` so we can clone it and change the color |
| editorField | So we can get access to the object the holds the cursor drawable | 
| cursorDrawableField | So we can update that with the new drawable we created | 

I can go through and add more step by step inline documentation if you think that will help.  

Also worth noting starting in API 28, Google restricts the use of reflection. So this won't work in API 28 and above. Once we update Aztec to target API 29 though we can set the drawable using [setTextCursorDrawable](https://developer.android.com/reference/android/widget/TextView#setTextCursorDrawable)

Reply:
Thanks for the explanation. We need to make sure the screen reader doesn't say weird things after making these fields accessible. Could you be able to set selectionColor on a RichText and test accessibility/screen reader? Thanks!

Reply:
> API 29 supports setTextCursorDrawable which would be a cleaner way to handle this when an upgrade to that API level occurs.

Also, could you open a tech debt task in this repo and add it to our board to help us remember this in the future?

Reply:
> Could you be able to set selectionColor on a RichText and test accessibility/screen reader?

I tested this on my Pixel 3a, which runs API 29, and an API 24 emulator after installing https://play.google.com/store/apps/details?id=com.google.android.marvin.talkback&hl=en_US in the emulator. From my testing, it's behaving the same both before and after this change. 

> could you open a tech debt task

Just created that here: https://github.com/wordpress-mobile/gutenberg-mobile/issues/1787